src/Apps/Order/Components/ShippingAndPaymentDetails.tsx

```diff
@@ -1,38 +1,107 @@
 import React from "react"
 
 import { Serif, space } from "@artsy/palette"
+import { ShippingAndPaymentDetails_order } from "__generated__/ShippingAndPaymentDetails_order.graphql"
+import { createFragmentContainer, graphql } from "react-relay"
 import { StackableBorderBox } from "Styleguide/Elements/Box"
-import { Flex } from "Styleguide/Elements/Flex"
-import {
-  CreditCardIcon,
-  CreditCardType,
-} from "Styleguide/Elements/icons/CreditCardIcon"
+import { Flex, FlexProps } from "Styleguide/Elements/Flex"
+import { CreditCardIcon } from "Styleguide/Elements/icons/CreditCardIcon"
 
-interface ShippingAndPaymentDetailsProps {
-  address: string
-  creditCardLast4digits: string
-  creditCardtype: CreditCardType
-  creditCardExpiry: string
+interface ShippingAndPaymentDetailsProps extends FlexProps {
+  order: ShippingAndPaymentDetails_order
+}
+
+const renderAddress = ({
+  shippingName,
+  shippingAddressLine1,
+  shippingAddressLine2,
+  shippingCity,
+  shippingPostalCode,
+  shippingRegion,
+}: {
+  shippingName: string
+  shippingAddressLine1: string
+  shippingAddressLine2: string | null
+  shippingCity: string
+  shippingPostalCode: string | null
+  shippingRegion: string | null
```

Comment:
Might these types be better served as an interface?

Reply:
I wonder if maybe the `Order` type should have a `fulfillment` field which returns a union of a `Pickup` or a `Shipping` type. That way the `Order` type doesn’t get overloaded with all these `shippingFoo` fields _and_ when e.g. the type is `Pickup` you don’t end up with all these `shippingFoo` fields in the response data.

Reply:
interesting point, im not too familiar with union type in GraphQL, MP work here is pretty minimal and mostly copies Exchange's schema, if we want to do this i'd suggest lets do it in Exchange and we can use/copy it here in MP too.

Reply:
(This does not block this PR btw.)

Reply:
https://github.com/artsy/exchange/pull/119